browser/src/libs/github/util.tsx

```diff
@@ -182,29 +182,44 @@ function getDiffResolvedRevFromPageSource(pageSource: string, isPullRequest: boo
 }
 
 /**
- * Scrapes the branch name from the branch select menu element
- * present on GitHub blob pages.
+ * Returns the file path for the current page. Must be on a blob page.
+ *
+ * Implementation details:
+ *
+ * - This scrapes the file path from the permalink on GitHub blob pages:
+ *
+ *     <a class="d-none js-permalink-shortcut" data-hotkey="y" href="/gorilla/mux/blob/ed099d42384823742bba0bf9a72b53b55c9e2e38/mux.go">Permalink</a>
+ *
+ * - We can't get the file path from the URL because the branch name can contain
+ *   slashes which make the boundary between the branch name and file path
+ *   ambiguous. For example:
+ *
+ *     https://github.com/sourcegraph/sourcegraph/blob/bext/release/cmd/frontend/internal/session/session.go
+ *
+ * TODO ideally, this should only scrape the code view itself.
  */
-export function getBranchName(): string {
-    const branchSelectMenu = document.querySelector('.branch-select-menu') as HTMLElement
-    if (!branchSelectMenu) {
-        throw new Error('Could not find .branch-select-menu')
-    }
-    const cssTruncateTarget = branchSelectMenu.querySelector('span.css-truncate-target') as HTMLElement
-    if (!cssTruncateTarget) {
-        throw new Error('Could not find span.css-truncate-target')
-    }
-    if (!cssTruncateTarget.innerText.endsWith('…')) {
-        // The branch name is not truncated
-        return cssTruncateTarget.innerText
-    }
-    // When the branch name is truncated, it is stored in full
-    // in the select menu button's title attribute.
-    const selectButton = cssTruncateTarget.closest('.select-menu-button') as HTMLElement
-    if (!selectButton) {
-        throw new Error('Could not find .select-menu-button')
-    }
-    return selectButton.title
+export function getFilePath(): string | null {
+    const permalink = document.querySelector('.js-permalink-shortcut')
+    if (!permalink) {
+        console.error(`Unable to determine the file path because no .js-permalink-shortcut element was found.`)
+        return null
+    }
+    const href = permalink.getAttribute('href')
+    if (!href) {
+        console.error(
+            `Unable to determine the file path because the .js-permalink-shortcut element did not have an href attribute.`
+        )
+        return null
+    }
+    // @ts-ignore these unused variables aid readability
+    const [_, _user, _repo, _blob, _commitID, ...path] = href.split('/')
```

Comment:
`@ts-ignore` is _really_ dangerous because it disables _any_ TypeScript error, even type errors. For example, you could change the type of `href` to something other than string here and not get a compile error anymore for calling `split()` on it.

If it's only for documentation, use comments:
```suggestion
    const [, /* user */, /* repo */, /* blob */, /* commitID */, ...path] = href.split('/')
```

Reply:
I would also suggest parsing the `href` with `URL` first and only running this logic on `url.pathname` to be resilient against missing protocol `//`, hash fragments (line number?), search queries (unified/split?) etc

Reply:
Avoided `@ts-ignore` :+1: Unfortunately, prettier does this:

```typescript
    const [, , , , , /* user */ /* repo */ /* blob */ /* commit ID */ ...path] = href.split('/')
```

So I went with this instead:

```typescript
    // <empty>/<user>/<repo>/blob/<commitID>/<path/to/file>
    const [, , , , , ...path] = href.split('/')
```

Reply:
The attribute looks like `href="/user/repo/bob/commitID/path/to/file"`, and that's not a valid URL, so I can't use `URL` to parse it. However, I found out that there's a `.href` attribute on anchor tags with the full URL, so I used that in order to safeguard against GitHub changing the href attribute to a full URL at some point.

Reply:
`URL` can parse relative URLs like `new URL(a.href, window.location.href)`

Reply:
Neat! I'll stick to `.href` because it's `string`, whereas `getAttribute('href'): string | null`.